File: euler_026.py
from time import time

def unit_fraction(n):
    num = 10**(len(str(n)))
    while True:
        rem = num//n
        yield (num,rem)
        num = (num - rem*n)
        num *= 10

def recurring_length(n):
    uf = unit_fraction(n)
    nums = []
    while True:
        u_step = next(uf)
        if u_step[0] == 0:
            return 0
        if u_step[0] in nums:
            return len(nums)
        else:
            nums.append(u_step[0])

start = time()

max_length = 0
max_d = 0

for d in range(1,1000):
    l = recurring_length(d)
    if l > max_length:
        [max_length, max_d] = [l, d]

print("d with the longest recurring cycle: " + str(max_d) + " (length = " + str(max_length) + ")")

stop = time()
print("Time: " + str(stop-start))

File: euler_026.js
const start = Date.now();

let biggest = 0;
let d = 2

function unitFractionDecimals(d) {
    let table = {}
    // Initialization
    table["length"] = 0;
    table["dividend"] = 1;
    table["quotient"] = 0;
    table["remainders"] = new Set();
    table["recurring"] = false;
    table["string"] = "";

    while (table["dividend"] < d) {
        table["dividend"] *= 10;
    }

    while (!table["recurring"]) {
        table["quotient"] = Math.floor( table["dividend"] / d );
        let rem = table["dividend"] % d;
        if (rem === 0) {
            return 0;
        }
        if (table["remainders"].has(rem)) {
            table["recurring"] = true;
        }
        table["remainders"].add(rem);
        table["length"]++;
        table["dividend"] = 10 * rem;
        table["string"] += table["quotient"].toString();
    }

    return table["string"].length - 1;
}
    
for (let i = 3; i < 1000; i++) {
    let len = unitFractionDecimals(i);
    if ( len > biggest) {
        d = i;
        biggest = len;
    }
}

const end = Date.now();

console.log("Execute time: ", end - start, " Value with longest recurring cycle is: ", d);